x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_data_streams/index.ts

```diff
@@ -16,11 +17,11 @@ export async function getDataStreams(options: {
   datasetQuery?: string;
   uncategorisedOnly: boolean;
 }) {
-  const { esClient, type, datasetQuery, uncategorisedOnly } = options;
+  const { esClient, type = DEFAULT_DATASET_TYPE, datasetQuery, uncategorisedOnly } = options;
 
   const datasetName = streamPartsToIndexPattern({
-    typePattern: type ?? '*',
-    datasetPattern: datasetQuery ? `*${datasetQuery}*` : '*',
+    typePattern: type,
+    datasetPattern: datasetQuery ? `*${datasetQuery}*` : '*-*',
```

Comment:
Is that intentional to allow for matching against the namespace too? Or should this be `*${datasetQuery}*-*`? If so, would it make sense to use `dataStreamPartsToIndexName` instead?

Reply:
Here we are targeting `namespaces` only if we don't get the `datasetQuery`, in a practical example this would look like `logs-*-*` if no datasetQuery is set up

Reply:
Right - but if the user enters `abc-def`, wouldn't it target any dataset ending with `abc` and namespace starting with `def`?

Reply:
yes, that's the case, I think that's aligned with the initial plan but now we are simply filtering in the UI. Is your suggestion to use `datasetQuery` instead of `*datasetQuery*`?

Reply:
Intuitively I would have expected this to use `*{datasetQuery}*-*` to maintain the `${type}-*-*` pattern this issue is about. But I guess it's not really that important in this case.